src/Voron/StorageEnvironmentOptions.cs

```diff
@@ -471,6 +473,20 @@ protected override void Disposing()
                     if (journal.Value.IsValueCreated)
                         journal.Value.Value.Dispose();
                 }
+
+                lock (_journalsForReuse)
+                {
+                    foreach (var reusableFile in _journalsForReuse.Values)
```

Comment:
You are deleting the recycled journals on shutdown. But is that something we want? 
If we restart, we will lose them.

On the other hand, that means that we don't take extra space for databases that have been shutdown...